devtools/cli/src/main/java/io/quarkus/cli/plugin/CliPluginsSync.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.cli.plugin;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.concurrent.Callable;
+
+import io.quarkus.cli.common.RunModeOption;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "sync", header = "Sync (discover / purge) CLI Plugins.")
```

Comment:
after doing `qss plugin add whatever` with nonexistent alias/name I kinda expected `qss plugin sync` to remove them or at least warn/complain about missing plugins.

Reply:
I think that `qss plugin add whatever` should not work in the first place.

`qss plugin sync` is meant to remove missing plugins that fall into the following categories:
- jbang alias
- binaries

`qss plugin add whatever` seems to add it as a jbang alias. 
So, yes we have two distinct issues here:
- plugin  should not be added
- sync should remove it.